### ISSUE TITLE
[bug][1.25.0 regression] 'NoneType' object has no attribute 'get_safe'

### ISSUE THREAD

**Issue by TheQwertiest** [conan-io/conan#7025]
### Environment Details (include every applicable attribute)
  * Operating System+version: Ubuntu 
  * Compiler+version: gcc 5.4.0 (stock/Linaro)
  * Conan version: 1.25.1
  * Python version: 3.8.2

### Steps to reproduce (Include if Applicable)
- Create a following recipe:
```py
from conans import ConanFile

from conans.client.build.autotools_environment import AutoToolsBuildEnvironment

class Pkg(ConanFile):
    name = 'conan_bug_pkg'
    version='1.0'
    exports_sources = ['*']

    def build(self):
        AutoToolsBuildEnvironment(self)

    def package(self):
        pass
```
- Create two profiles `native` (with stock compiler and arch) and `cross` (e.g. cross-toolchain and target arch `arm`). See attached profiles for examples.
- `create . @ --profile:build native --profile:host cross`

### Logs (Executed commands with output) (Include/Attach if Applicable)
```
ERROR: conan_bug_pkg/1.0: Error in build() method, line 11
	AutoToolsBuildEnvironment(self)
	AttributeError: 'NoneType' object has no attribute 'get_safe'
```

Note: this does not happen in 1.24.1.
I've backtraced this error to [this line](https://github.com/conan-io/conan/blob/ddf9cca9ea45879986adec8b0308a0519226bbac/conans/client/graph/graph_builder.py#L406), where `current_node.conanfile.settings_target` is equal to `None`, which leads to [this code](https://github.com/conan-io/conan/blob/63b6a28ab36706144dbdc0ddf40ead09522fecc1/conans/client/tools/oss.py#L579) failing.

Example profiles:
[native.txt](https://github.com/conan-io/conan/files/4633363/native.txt)
[arm.txt](https://github.com/conan-io/conan/files/4633361/arm.txt)

**Comment by TheQwertiest** [comment]
Related feature: https://github.com/conan-io/conan/pull/6769

**Comment by memsharded** [comment]
Hi @TheQwertiest 

Thanks very much for reporting this. Indeed a bug. Not strictly a regression, this feature is still experimental, but I have submitted a https://github.com/conan-io/conan/pull/7032 pull request, and targeted for a potential 1.25.2.

**Comment by jgsogo** [comment]
This will be released in 1.25.2.

Thanks!
